src/cryptonote_core/blockchain.cpp

```diff
@@ -1126,6 +1126,7 @@ bool Blockchain::switch_to_alternative_blockchain(std::list<blocks_ext_by_hash::
   }
 
   // if we're to keep the disconnected blocks, add them as alternates
+  const size_t discarded_blocks = disconnected_chain.size();
```

Comment:
Do we really want to keep discarded blocks? And then confuse daemon about correct chain?